go-controller/pkg/metrics/ovn.go

```diff
@@ -279,6 +284,115 @@ func ovnControllerConfigurationMetricsUpdater() {
 	}
 }
 
+func ovnControllerServiceSubnetMetricsPublisher() {
+	cidrV4, cidrV6, v4Size, v6Size := getSubnetDetails(config.Kubernetes.ServiceCIDRs)
+	var metricServiceSubnet = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: MetricOvnkubeNamespace,
+		Subsystem: MetricOvnkubeSubsystemMaster,
+		Name:      "service_subnet",
+		Help: "A metric with a constant '1' value labeled by subnet that " +
+			"specifies the subnet used for service. subnet_v4 and subnet_v6 are " +
+			"used as labels for IPv4 and IPv6 subnets respectively. If any of the subnets " +
+			"is not configured, corresponding label will have a blank value.",
+	},
+		[]string{
+			"subnet_v4",
+			"subnet_v6",
+		},
+	)
+	ovnRegistry.MustRegister(metricServiceSubnet)
+	metricServiceSubnet.WithLabelValues(cidrV4, cidrV6).Set(1)
+
+	ovnRegistry.MustRegister(prometheus.NewGaugeFunc(
+		prometheus.GaugeOpts{
+			Namespace: MetricOvnkubeNamespace,
+			Subsystem: MetricOvnkubeSubsystemMaster,
+			Name:      "service_subnet_size_ipv4",
+			Help:      "Size of IPv4 service subnet size. 0 for the value means IPv4 subnet is not configured.",
+		},
+		func() float64 {
+			return float64(v4Size)
+		}))
+
+	ovnRegistry.MustRegister(prometheus.NewGaugeFunc(
+		prometheus.GaugeOpts{
+			Namespace: MetricOvnkubeNamespace,
+			Subsystem: MetricOvnkubeSubsystemMaster,
+			Name:      "service_subnet_size_ipv6",
+			Help:      "Size of IPv6 service subnet size. 0 for the value means IPv6 subnet is not configured.",
+		},
+		func() float64 {
+			return float64(v6Size)
+		}))
+}
+
+func ovnControllerClusterSubnetMetricsPublisher() {
+	var cidrs []*net.IPNet
+
+	for _, subnet := range config.Default.ClusterSubnets {
+		cidrs = append(cidrs, subnet.CIDR)
+	}
+	cidrV4, cidrV6, v4Size, v6Size := getSubnetDetails(cidrs)
+
+	var metricClusterSubnet = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: MetricOvnkubeNamespace,
+		Subsystem: MetricOvnkubeSubsystemMaster,
+		Name:      "cluster_subnet",
```

Comment:
`s/cluster_subnet/cluster_subnets`?

Reply:
We do support multiple cluster CIDRs. How will we show those in metrics? the `getSubnetDetails` will only show the subnet sizes for the last subnet cidr of each type v4 and v6 as it is coded up right now.

Reply:
I had comma separated values for this which I will revert to. Once this is done please review and let me know.